components/wrap/WrapModal.tsx

```diff
@@ -0,0 +1,140 @@
+import * as React from "react";
+import Modal from "react-bootstrap/Modal";
+import Button from "react-bootstrap/Button";
+import Form from "react-bootstrap/Form";
+import Row from "react-bootstrap/Row";
+import Col from "react-bootstrap/Col";
+import { ethers } from "ethers";
+import { ethxABI, NETWORK_ID, PAYMENT_TOKEN_ADDRESS } from "../../lib/constants";
+import { getETHBalance, getETHxBalance } from "../../lib/getBalance";
+
+function WrapMpdal({ account, signer, show, handleClose }: any) {
+  const [ETHBalance, setETHBalance] = React.useState<string | undefined>();
+  const [ETHxBalance, setETHxBalance] = React.useState<string | undefined>();
+  const [isWrapping, setIsWrapping] = React.useState<boolean>(false);
+  const [isBalanceInsufficient, setIsBalanceInsufficient] = React.useState<boolean>(false);
+
+  React.useEffect(() => {
+    let isMounted = true;
+
+    (async () => {
+      const provider = new ethers.providers.InfuraProvider(
+        NETWORK_ID,
+        process.env.NEXT_PUBLIC_INFURA_ID
+      );
+
+      const ethBalance = await getETHBalance(provider, account);
+
+      const ethxBalance = await getETHxBalance(provider, account);
+
+      if (isMounted) {
+        setETHBalance(ethBalance);
+        setETHxBalance(ethxBalance);
+      }
+    })();
+
+    return () => {
+      isMounted = false;
+    };
+  }, []);
+
+  const wrapETH = async (amount: string) => {
+    const provider = new ethers.providers.InfuraProvider(
+      NETWORK_ID,
+      process.env.NEXT_PUBLIC_INFURA_ID
+    );
+      
+    const ETHx = new ethers.Contract(PAYMENT_TOKEN_ADDRESS, ethxABI, provider);
+
+    const reciept = await ETHx.connect(signer).upgradeByETH({
+      value: ethers.utils.parseEther(amount)
+    });
+
+    setIsWrapping(true);
+
+    try {
+      await reciept.wait();
+      // Wrap ETHx successfully!
+      const ethBalance = await getETHBalance(provider, account);
+      const ethxBalance = await getETHxBalance(provider, account);
+      // Update balances
+      setETHBalance(ethBalance);
+      setETHxBalance(ethxBalance);
+    } catch (error) {
+      console.error(error);
+    };
+
+    setIsWrapping(false);
+  }
+
+  const onSubmit = (e: any) => {
+    e.preventDefault();
+    const amount = Number((e.target[0].value));
+
+    if (amount > Number(ETHBalance)) {
+      setIsBalanceInsufficient(true);
+    } else {
+      setIsBalanceInsufficient(false);
+
+      if (!Number.isNaN(amount) && amount > 0) {
+        console.log("amount is valid", amount);
+        (async () => {
+          await wrapETH(amount.toString());
+        })();
+      }
+    }
+  }
+
+  return (
+    <Modal
```

Comment:
Could you add some styling to the modal? See `ProfileModel` for an example. There should just be some bootstrap classes that need to be added.

Reply:
The styling is much better. One last thing: could you update the style of the text field to be like the others? And also have the wrap button be either much closer or off to the side?

Sorry about not having very specific specs in regards to design and layout 😔